conformance/Makefile

```diff
@@ -2,7 +2,6 @@ NKG_TAG = edge
 NKG_PREFIX = nginx-kubernetes-gateway
 GATEWAY_CLASS = nginx
 SUPPORTED_FEATURES = HTTPRoute,HTTPRouteQueryParamMatching,HTTPRouteMethodMatching,HTTPRoutePortRedirect,HTTPRouteSchemeRedirect
-EXEMPT_FEATURES = ReferenceGrant
```

Comment:
It only applies to features, not individual tests. Since ReferenceGrant is the only feature we need to exclude, I figured I would just remove the variable. I can add it back if you think it will still be useful.